### PR TITLE
Feat: add a functional overlay

### PR DESCRIPTION
**Pull request by fitzhavey**
Inspired by the changes on [feat/overlay-2.0](https://github.com/holistic-web/ar-examples/tree/feat/overlay-2.0), this PR adds a simple overlay that we can add to all our components.

# Noteworthy things:
- brought over the changes from [feat/deeper-vue](https://github.com/holistic-web/ar-examples/tree/feat/deeper-vue) so that we no longer need to load the page when navigating
- attempted to set single page application to render as fullscreen, doesn't seem to work - but might when properly hosted
- adds a default layout component, with an auto generated header and slots for controls and content
- added [bootstrap-vue](https://bootstrap-vue.org/docs/components), giving us some handy UI components
- added instructions / controls to each demo
- modified the "flapping-birds" demo to no longer register a custom component, but instead dynamically add birds to the scene based given inputs

I've added this PR's preview URL to the list of authorised URLs in 8th wall, so we can preview the changes here:
https://augmented-reality--pr2-feat-functional-over-16vjzxiu.web.app/

### CODE REVIEW

**Comment by subin-j** [comment]
wat is dis

**Comment by fitzhavey** [comment]
@subin-j it's a few things I've been trying to tie together, to use as building blocks for some AR gimmicks / games and to hopefully be more fluent in webAR tech if / when it becomes more mainstream.

Trying to be as brief as possible:
- it uses Google / Niantic's [8th wall](https://www.8thwall.com/) webAR framework, which includes a more polished version of Mozilla's [AFrame](https://aframe.io/) to build mobile AR experiences
- it's hosted on [firebase](https://firebase.google.com/) after it inherited the url from [my sandbox](https://github.com/michaelfitzhavey/sandbox), which is how we get the shiny `https://augmented-reality.web.app` url
- developed through github codespaces to allow hot reloading builds that are served over https (required since you can't test on localhost since you need access to the camera)
    _codespaces is sweet, I was let in ~1 hour after signup and it makes life so easy after syncing vscode settings, so give it a shot! https://github.com/features/codespaces/signup_
- All of the above is bundled up in [nuxtjs](https://nuxtjs.org/), so we get the benefits of a fully featured JS component framework, NPM and compilation / linting, though we still build as a single static site.

If you want to talk through it sometime, or just have some ideas you think would look cool in augmented reality, give me a shout 🚀 

TL;DR I guess I'm trying to build the toolset for more creative people to make very sweet AR games 👾

**Comment by fitzhavey** [comment]
8 days since review [request], going to just merge this one. To a nicer world (somehow)! 🍻
